File: src/test/java/destiny/core/chinese/ziwei/StarGeneralFrontTest.java
/**
 * Created by smallufo on 2017-06-16.
 */
package destiny.core.chinese.ziwei;

import org.junit.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Locale;

import static destiny.core.chinese.Branch.戌;
import static destiny.core.chinese.Branch.申;
import static destiny.core.chinese.Branch.酉;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertSame;

public class StarGeneralFrontTest {

  private Logger logger = LoggerFactory.getLogger(getClass());

  @Test
  public void testToString() {
    for(StarGeneralFront star : StarGeneralFront.values) {
      assertNotNull(star.toString());
      assertNotNull(star.toString(Locale.TAIWAN));
      assertNotNull(star.toString(Locale.CHINA));
      logger.info("tw = {}({}) , cn = {}({})" ,
        star.toString(Locale.TAIWAN) , star.getAbbreviation(Locale.TAIWAN),
        star.toString(Locale.CHINA) , star.getAbbreviation(Locale.CHINA));
    }
  }

  @Test
  public void testFun() {
    // 酉年，將星[1] 在酉 , 攀鞍[2] 在戌 , 亡神[12] 在 申
    assertSame(酉 , StarGeneralFront.fun將星.apply(酉));
    assertSame(戌 , StarGeneralFront.fun攀鞍.apply(酉));
    assertSame(申 , StarGeneralFront.fun亡神.apply(申));
  }
}

File: src/test/kotlin/destiny/core/chinese/ziwei/StarGeneralFrontTest.kt
/**
 * Created by smallufo on 2017-06-16.
 */
package destiny.core.chinese.ziwei

import destiny.core.AbstractPointTest
import destiny.core.chinese.Branch.*
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.Json.Default.decodeFromString
import kotlin.test.Test
import kotlin.test.assertSame

class StarGeneralFrontTest : AbstractPointTest(StarGeneralFront::class) {

  @Test
  fun testSerialize() {
    StarGeneralFront.values.forEach { s ->
      val rawJson = Json.encodeToString(s)
      logger.info { "$s = $rawJson" }
      assertSame(s, decodeFromString(rawJson))
    }
  }

  @Test
  fun testFun() {
    // 酉年，將星[1] 在酉 , 攀鞍[2] 在戌 , 亡神[12] 在 申
    assertSame(酉, StarGeneralFront.fun將星.invoke(酉))
    assertSame(戌, StarGeneralFront.fun攀鞍.invoke(酉))
    assertSame(申, StarGeneralFront.fun亡神.invoke(酉))
  }
}
